### PR TITLE
dev tools: deploy to local validator with Jakefile / rhopm

### LINKED ISSUE #17: local valiator dev tools (with registry support)
IOU a copy of my klunky Makefile, but I'd rather not inflict it on any / many more devs.

See also:

https://github.com/rchain-community/rchain-docker-shard

@tgrospic adds:
> in client js example is also compose file (with bonds and wallets) for local testing.
https://github.com/tgrospic/rnode-client-js#run-rnode-with-docker

### PR DESCRIPTION
**Pull request by dckc**
fixes #17

### CODE REVIEW

File: rclient/types.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  @typedef { {
+    adminBase(): string,
+    propose(): Promise<string>
+  } } RNodeAdmin
+
+  @typedef { {
+    setTimeout: typeof setTimeout,
+    clearTimeout: typeof clearTimeout,
+  } } SchedulerAccess
```
Comment (tgrospic) [inline rclient/types.js:95]: Kudos for writing this by hand!

File: rclient/types.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+/**
+ * ref
+ * https://github.com/rchain/rchain/blob/dev/node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala
+ * https://github.com/rchain/rchain/blob/dev/node/src/main/scala/coop/rchain/node/web/AdminWebApiRoutes.scala
+ * https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala
+ * https://github.com/rchain/rchain/blob/dev/node/src/main/scala/coop/rchain/node/api/WebApi.scala#L120
+ *
+ * @typedef {{
+ *    term: string,
+ *    timestamp: number, // milliseconds
```
Comment (tgrospic) [inline rclient/types.js:10]: To be completely correct, timestamp can be any number (int64) unrelated to milliseconds.

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1 @@
+../agoric/agoric-sdk/packages/xs-vat-worker/.eslintrc.js
```
Comment (tgrospic) [inline .eslintrc.js:1]: Hm. Is this file meant to be inside project?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+```
+npm install
+npx jake
+```
+
+## Automated deployment with jake
+
+[Jake](https://jakejs.com/) is a JavaScript build tool for NodeJS, much like `make`.
+_Recall that `npx` is a mechanism for launching executables in npm project dependencies,
+i.e. in `node_modules/.bin`._
```
[the remark this answers is not in the document]
Comment (dckc) [reply CONTRIBUTING.md:14]: I had it as a `devDependency` but eslint griped at me. Then I thought: well, running the Jakefile is what this package is all about; the Jakefile isn't just a build tool in service to something else here.

(The fact that npx downloads and runs things without installation is kinda scary, but I put up with it for now.)

File: Jakefile.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ */
+// @ts-check
+
+'use strict';
+
+const io = {
+  sched: { setTimeout, clearTimeout, setInterval, clearInterval },
+  clock: () => new Date(),
+  // eslint-disable-next-line global-require
+  fs: require('fs'),
```
[the remark this answers is not in the document]
Comment (dckc) [reply Jakefile.js:15]: I think `readFile` and `writeFile` (from `fs.promises`) are also used, no? But yes, it's good to reduce the API surface that gets imported.

File: rclient/rnode.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  }
+}
+
+/**
+ * @param {typeof fetch} fetch
+ */
+export function RNode(fetch) {
+  /**
+   * @param {string} methodUrl
+   * @param {?Object} request
```
[the remark this answers is not in the document]
Comment (dckc) [reply rclient/rnode.js:17]: multiple bottom values in JS makes me sad. I clearly don't like to think about it.

File: .eslintrc.js (hunk 3 above)
Comment (dckc) [reply .eslintrc.js:1]: I meant to copy it; did I make a symlink?!

File: rclient/types.js (hunk 2 above)
Comment (dckc) [reply rclient/types.js:10]: oh... right... validAfterBlockNumber is the part that provides constraint w.r.t. real time

File: Jakefile.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (tgrospic) [reply Jakefile.js:15]: You are right, I didn't see the use in rhopm.js.

**Comment by dckc** [comment]
Thanks for the review, @tgrospic . There are things I'd like to improve, but this is a vast improvement over current master, so I think it's time to merge. Note that most of the `rhopm` stuff has moved to https://github.com/rchain-community/rchain-api/tree/http-client ; what's left here is mostly just `Jakefile.js`.

**Pull request merged**